### ISSUE TITLE
Key is re-triggered when other key is let go

### ISSUE THREAD

**Issue by maxbergmark** [gaucho-labs/leptos-hotkeys#119]
Hey! I've been using this crate in a new project, and everything is working great except for one small issue. This issue can be reproduced on the demo page.

![image](https://github.com/user-attachments/assets/fafddcf5-4bbb-4404-8f3a-b9cdc1505c56)

## Steps to reproduce
1. Press and hold down arrow up (Count goes up by 1)
2. Quickly press and release arrow right (Count goes up by 2)
3. Press arrow right (Count goes up by 1)
4. Release arrow right (Count goes up by 1)
5. Release arrow up (Count is unchanged)

It seems as if the up arrow event is triggered when the right arrow is released, as long as the up arrow is held down. I have also noticed that the effect is quite inconsistent depending on which arrow key is held down, and which is pressed after. 

Another way to visualize the behavior can be seen on [my Leptos Sudoku game/solver](https://maxbergmark.github.io/sudoku_solver_wasm/?sudoku=1.4.2.e7.b5826.g3.d2.b3.9.j4.5.e6.c64.b5.c8.7.c2.f17.). If you click a square there, you are able to change your selected square using the arrow keys. But if you accidentally press two arrow keys at the same time, the results are not as expected.

**Comment by friendlymatthew** [comment]
Thanks for creating an issue @maxbergmark! FYI -- work's been ramping up and it'll take me some time to get to this... 

Feel free to open a PR and try attacking this issue! (The entire codebase isn't that big either!)

> It seems as if the up arrow event is triggered when the right arrow is released, as long as the up arrow is held down. I have also noticed that the effect is quite inconsistent depending on which arrow key is held down, and which is pressed after.

The demo actually binds `ArrowUp` or `ArrowRight` keys to increment the count. Same behavior using `ArrowDown` or `ArrowLeft` to decrement the count. I just never updated the demo to include that.
https://github.com/gaucho-labs/leptos-hotkeys/blob/4b977cf92b5b3801d7610c723f83239ef6daf804/examples/demo/src/app.rs#L78-L88


<br />
<br />

> Another way to visualize the behavior can be seen on [my Leptos Sudoku game/solver](https://maxbergmark.github.io/sudoku_solver_wasm/?sudoku=1.4.2.e7.b5826.g3.d2.b3.9.j4.5.e6.c64.b5.c8.7.c2.f17.). If you click a square there, you are able to change your selected square using the arrow keys. But if you accidentally press two arrow keys at the same time, the results are not as expected.

Ah -- this is bad (the undesired behavior). I'll have to spend more time investigating this and see where it went wrong. But once again, I'm pretty busy atm.


Also, the web app looks very sleek!

**Comment by maxbergmark** [comment]
I did some additional testing by enabling the "debug" option for the crate. From there, it is quite apparent what's happening. Here, I started by pressing arrow up, and then quickly pressed and released arrow right while arrow up was held. 
![image](https://github.com/user-attachments/assets/8d8b587a-de1e-49cd-b8d5-e8c0e5f32720)

The culprit seems to be this code in `use_hotkeys.rs`:
```rust
let mut pressed_keyset = pressed_keys.get();
if let Some(matching_hotkey) = parsed_keys
    .iter()
    .find(|hotkey| is_hotkey_match(hotkey, &mut pressed_keyset))
```
With how the signal `pressed_keys` is set up, it will fire events both on keyup and keydown events (both of course modify the underlying hashset). With the key presses described above, the following events are triggered:

1. KeyDown `{"ArrowUp"}`
2. KeyDown `{"ArrowUp", "ArrowRight"}`
3. KeyUp `{"ArrowUp"}`
4. KeyUp `{}`

Here, it is clear why "ArrowUp" is triggered three times, even if the button is only pressed once.

The easiest fix I could think of would be to also track the last pressed key separately from the hashset, and only trigger hotkey events that include the last pressed key.

**Comment by maxbergmark** [comment]
Quick update: I think I have a fix for the problem. I just replaced the existing hashmap containing all keypresses with a struct:
```rust
#[derive(Debug, Default, Clone)]
#[cfg_attr(feature = "ssr", allow(dead_code))]
pub struct KeyPresses {
    pub keys: std::collections::HashMap<String, web_sys::KeyboardEvent>,
    pub last_key: Option<String>,
}
```
I would like to test this more thoroughly before creating a PR, but I noticed that I don't have permission to push a new branch when I tried to push my changes to a bugfix branch.

**Comment by maxbergmark** [comment]
I created a PR, and tested the functionality locally using the sudoku app I linked above. From my testing, all of these things work as expected:
1. Single key presses
2. Pressing two separate key presses simultaneously
3. Pressing a key while another key is held down
4. Using a multi-key hotkey while one or more other keys are held down
5. Pressing a key while some multi-key hotkey is held down

I would really like to avoid breaking any existing applications using Leptos Hotkeys, so I would appreciate an extra set of eyes on the code changes. And if there's some refactoring needed before merging this, let me know!
